### PR TITLE
Audiencelogy Bid Adapter : Initial Release

### PR DESCRIPTION
**Pull request by hasanideepak**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

File: modules/audiencelogyBidAdapter.js [hunk 1]
```diff
@@ -123,41 +130,32 @@ function fetchSite(bidderRequest) {
...
-    creativeId: bid && bid.crid ? bid.crid : undefined,
-    meta: {
-      advertiserDomains: bid && bid.adomain ? bid.adomain : []
-    },
-    netRevenue: false,
-    currency: bid && bid.cur ? bid.cur : 'USD',
-    dealId: bid && bid.dealId ? bid.dealId : undefined,
-    ttl: 300
+const buildResponse = (bid) => {
+  let response = {};
```
Comment (patmmccann) [inline modules/audiencelogyBidAdapter.js:134]: We can't accept changes designed to fool the duplication checker

**Review by patmmccann — CHANGES_REQUESTED** [review]
Please import common functions from a library

File: modules/audiencelogyBidAdapter.js (hunk 1 above)
Comment (hasanideepak) [reply modules/audiencelogyBidAdapter.js:134]: I understand, please let me know where to add common functions in library?

**Comment by ChrisHuie** [comment]
Please add docs to -> https://github.com/prebid/prebid.github.io

**Comment by hasanideepak** [comment]
> Please add docs to -> https://github.com/prebid/prebid.github.io

Added https://github.com/prebid/prebid.github.io/pull/5457

File: modules/audiencelogyBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/audiencelogyBidAdapter.js:134]: Sure! Thank you. See #11854 for example or #11879

File: modules/audiencelogyBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  response.currency = bid && bid.cur ? bid.cur : 'USD';
+  response.dealId = bid && bid.dealId ? bid.dealId : undefined;
+  response.ttl = 300;
+  return response;
+}
+// Function to build the user object
+const setUser = (bid) => {
+  let user = {};
+  if (bid && bid.params) {
+    user.buyeruid = localStorage.getItem('adx_profile_guid') ? localStorage.getItem('adx_profile_guid') : '';
```
Comment (patmmccann) [inline modules/audiencelogyBidAdapter.js:154]: You have to import the storage manager to do this, publishers manage bidder access to storage depending on consent or their preferences

File: modules/audiencelogyBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        logError(err);
+      }
+    }
+    return resp;
+  }
+}
+// Function to fetch bid_floor
+const fetchFloor = (bid) => {
+  if (bid.params && bid.params.bid_floor) {
+    return bid.params.bid_floor;
```
Comment (patmmccann) [inline modules/audiencelogyBidAdapter.js:97]: Make sure to check the getFloor function from the request

**Review by patmmccann — CHANGES_REQUESTED** [review]
Looks pretty good, couple of tweaks and use of an import please

**Comment by hasanideepak** [comment]
@patmmccann I have added library audiencelogyUtils/bidderUtils.js and imported common code. Please verify.

**Comment by patmmccann** [comment]
@hasanideepak yes but you should also import in relevate, which has the same concern above, it uses local storage without importing storage manager

**Comment by patmmccann** [comment]
>  I will need to add same library in relevate fork in order to use it, doesnt that give conflict error as same code will be there in audiencelogy and relevate?

both will import it, but the build will be more compact as the code will only be in the project once. It will clear the error. 

> tried using storage manager, but it was unable to read the value from local storage. Can you please let me know some example.

https://github.com/prebid/Prebid.js/blob/ca24c395555167e5aba7007a83332c1d49441921/modules/ixBidAdapter.js#L110

What you have in relevate is a major rules violation, as it ignores consent. We need you to take care of it in this commit. I'll go ahead and submit a bug fix to delete what you are doing there now, so you'll want to merge in master after that merges to avoid merge conflicts

**Comment by hasanideepak** [comment]
> > I will need to add same library in relevate fork in order to use it, doesnt that give conflict error as same code will be there in audiencelogy and relevate?
> 
> both will import it, but the build will be more compact as the code will only be in the project once. It will clear the error.
> 
> > tried using storage manager, but it was unable to read the value from local storage. Can you please let me know some example.
> 
> https://github.com/prebid/Prebid.js/blob/ca24c395555167e5aba7007a83332c1d49441921/modules/ixBidAdapter.js#L110
> 
> What you have in relevate is a major rules violation, as it ignores consent. We need you to take care of it in this commit. I'll go ahead and submit a bug fix to delete what you are doing there now, so you'll want to merge in master after that merges to avoid merge conflicts

@patmmccann Thank you for the support. So, I will make the changes for store manager in this PR (i.e.audiencelogy) and create a new PR for relevate adapter to import common library and store manager changes. Is this right ? 
Merging both PR wont cause any conflicts ?

**Comment by patmmccann** [comment]
Just do it all in one?

**Comment by hasanideepak** [comment]
@patmmccann Considering we are an ad tech service provider to relevate and we also need our own prebid adapter, we would like to focus on relevate adapter at this moment to make it fully compliant with your policies (with storage manager changes), hence we would like to pause work on our own adapter (i.e. audiencelogy) at this moment and we will revisit in a few weeks.
Thank you for your support

**Comment by patmmccann** [comment]
No worries, just import stage manager instead of directly accessing local storage, keeping an eye out for your commit

**Pull request closed without merging**